obspy/clients/fdsn/mass_downloader/__init__.py

```diff
@@ -0,0 +1,503 @@
+
+Storing MiniSEED waveforms
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The MiniSEED storage rules are set by the ``mseed_storage`` argument of the
+:meth:`~obspy.clients.fdsn.mass_downloader.mass_downloader.MassDownloader.download`
+method of the
+:class:`~obspy.clients.fdsn.mass_downloader.mass_downloader.MassDownloader`
+class
+
+
+**Option 1: Folder Name**
+
+In the simplest case it is just a folder name:
+
+>>> mseed_storage = "waveforms"
+
+This will cause all MiniSEED files to be stored as
+
+``waveforms/NETWORK.STATION.LOCATION.CHANNEL__STARTTIME__ENDTIME.mseed``.
+
+An example of this is
+
+``waveforms/BW.FURT..BHZ__2014-10-27T16-37-23Z__2014-10-27T16-37-33Z.mseed``
+
+which is rather general but also quite long.
+
+**Option 2: String Template**
+
+For more control use the second possibility and provide a string containing
+``{network}``, ``{station}``, ``{location}``, ``{channel}``, ``{starttime}``,
+and ``{endtime}`` format specifiers. These values will be interpolated to
+acquire the final filename. The start and end times will be formatted with
+``strftime()`` with the specifier ``'%Y-%m-%dT%H-%M-%SZ'`` in an effort to
+avoid colons which are troublesome in filenames on many systems.
+
+>>> mseed_storage = ("some_folder/{network}/{station}/"
+...                  "{location}.{channel}.{starttime}.{endtime}.mseed")
+
+results in
+
+``some_folder/BW/FURT/.BHZ.2014-10-27T16-37-23Z.2014-10-27T16-37-33Z.mseed``.
+
+The download helpers will create any non-existing folders along the path.
+
+**Option 3: Custom Function**
+
+The most complex but also most powerful possibility is to use a function which
+will be evaluated to determine the filename. **If the function returns**
+``True`` **, the MiniSEED file is assumed to already be available and will not
+be downloaded again; keep in mind that in that case no station data will be
+downloaded for that channel.** If it returns a string, the MiniSEED file will
+be saved to that path. Utilize closures to use any other parameters in the
+function. This hypothetical function checks if the file is already in a
+database and otherwise returns a string which will be interpreted as a
+filename.
+
+>>> def get_mseed_storage(network, station, location, channel, starttime,
+...                       endtime):
+...     # Returning True means that neither the data nor the StationXML file
+...     # will be downloaded.
+...     if is_in_db(network, station, location, channel, starttime, endtime):
+...         return True
+...     # If a string is returned the file will be saved in that location.
+...     return os.path.join(ROOT, "%s.%s.%s.%s.mseed." % (network, station,
+...                                                       location, channel))
+>>> mseed_storage = get_mseed_storage
+
+.. note::
+
+    No matter which approach is chosen, if a file already exists, it will not
+    be overwritten; it will be parsed and the download helper class will
```

Comment:
Okay, probably this note needs a small adjustment then, since there are actually 3 conditions (filename+nonexistent, filename+existent, True) but it seems to imply there are only two (existent+nonexistent).